Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Internal/CoreRemoteMongoClient.swift

```diff
@@ -27,4 +33,36 @@ public class CoreRemoteMongoClient {
                                             withService: service,
                                             withDataSynchronizer: dataSynchronizer)
     }
+
+    private func onAuthEvent(_ authEvent: AuthRebindEvent) {
+        switch authEvent {
+        case .userRemoved(let removedUser):
+            let userId = removedUser.id
+            let key = appInfo.clientAppID + "/\(userId)"
+            try? CoreLocalMongoDBService.shared.deleteDatabase(
+                withKey: key,
+                withDBPath: "\(appInfo.dataDirectory.path)/local_mongodb/0/\(key)")
+        case .activeUserChanged(let currentActiveUser, _):
+            if lastActiveUserId != appInfo.authMonitor.activeUserId {
+                self.lastActiveUserId = appInfo.authMonitor.activeUserId
+                if currentActiveUser != nil {
+                    self.dataSynchronizer.reinitialize(appInfo: appInfo)
+                } else {
+                    self.dataSynchronizer.stop()
+                }
```

Comment:
When would currentActiveUser be null?

Reply:
If you log out, the active user changes from non-nil to nil. When you log out, we just want to stop the data synchronizer rather than reinitialize it

Reply:
Sorry,, saw this later on and realized it was a dumb question